test/DAICO.test.js

```diff
@@ -33,85 +33,188 @@ contract('DAICO', function (accounts) {
     await this.token.transfer(this.daico.address, tokenSupply);
   });
 
 
-  describe('withdraw', function () {
+  describe('withdrawal permissions and timing', function () {
 
 
-    it('owner can not withdraw before last withdrawal', async function () {
+    it('owner can withdraw after last withdrawal', async function () {
       let owner = accounts[0];
       let no_owner = accounts[3];
       await this.daico.buyTokens(no_owner, {from: no_owner, value: higher_value}).should.be.fulfilled;
       await increaseTimeTo(this.afterlastWithdrawn);
       await this.daico.withdraw({from: owner}).should.be.fulfilled;
     });
 
-    it('no one can withdraw if she is not owner', async function () {
-      let no_owner = accounts[3];
-      await this.daico.buyTokens(no_owner, {from: no_owner, value: higher_value}).should.be.fulfilled;
-      await increaseTimeTo(this.afterlastWithdrawn);
-      await this.daico.withdraw({from: no_owner}).should.be.rejectedWith(EVMRevert);
-    });
-
-    it('owner can withdraw after last withdrawal', async function () {
+    it('owner can not withdraw before last withdrawal', async function () {
       let owner = accounts[0];
       let no_owner = accounts[3];
       await this.daico.buyTokens(no_owner, {from: no_owner, value: higher_value}).should.be.fulfilled;
-      await increaseTimeTo(this.afterlastWithdrawn);
-      const initial_balance = web3.eth.getBalance(owner).c[0];
-      await this.daico.withdraw({from: owner}).should.be.fulfilled;
-      const final_balance = web3.eth.getBalance(owner).c[0];
-      assert.ok(final_balance > initial_balance);
+      await this.daico.withdraw({from: owner}).should.be.rejectedWith(EVMRevert);
     });
 
-    it('owner can t withdraw more than tap times time', async function () {
-      let owner = accounts[0];
+    it('no one can withdraw if she is not owner', async function () {
       let no_owner = accounts[3];
       await this.daico.buyTokens(no_owner, {from: no_owner, value: higher_value}).should.be.fulfilled;
       await increaseTimeTo(this.afterlastWithdrawn);
-      const initial_balance = web3.eth.getBalance(owner).c[0];
-      const timestamp = web3.eth.getBlock(web3.eth.blockNumber).timestamp;
-      await this.daico.withdraw({from: owner});
-      const final_balance = web3.eth.getBalance(owner).c[0];
-      assert.ok(final_balance - initial_balance <= tap * (timestamp - this.lastWithdrawn))
+      await this.daico.withdraw({from: no_owner}).should.be.rejectedWith(EVMRevert);
     });
+
   });
 
-  describe('voting as expected', function () {
+  describe('proposals', function () {
     it('no purchaser no holder', async function() {
       let purchaser = accounts[5];
       let isholder = await this.daico.isHolder(purchaser);
       assert.ok(!isholder);
     });
 
-    it('if yoy did not purchased a token you are not a holder', async function() {
+    it('if you did not purchased a token you are not a holder', async function() {
```

Comment:
should be `purchasers are considered holders` right?

Reply:
done